tests/test_vm.py

```diff
@@ -0,0 +1,93 @@
+import pytest
+from dotenv import load_dotenv
+import os
+from src.ssh_client import SSHClient
+from src.winrm_client import WinRMClient
+from src.utils import calculate_checksum
+
+load_dotenv()
+
+@pytest.fixture
+def shell_client(request):
+    os_option = request.param
+    if os_option == 'linux':
+        host = os.getenv('LINUX_HOST')
+        user = os.getenv('LINUX_USER')
+        password = os.getenv('LINUX_PASSWORD')
+        return SSHClient(host, user, password)
+    elif os_option == 'windows':
+        host = os.getenv('WINDOWS_HOST')
+        user = os.getenv('WINDOWS_USER')
+        password = os.getenv('WINDOWS_PASSWORD')
+        return WinRMClient(host, user, password)
+    else:
+        pytest.fail("Please provide a valid OS option: --os=linux or --os=windows")
+
+@pytest.fixture
+def directory(request):
+    dirpath = request.config.getoption("--dirpath")
+    if not dirpath:
+        pytest.fail("Please provide a directory path using --dirpath option.")
+    return dirpath
+
+@pytest.mark.parametrize("shell_client", ["linux", "windows"], indirect=True)
+def test_create_directory(shell_client, directory, request):
+    os_option = 'linux' if isinstance(shell_client, SSHClient) else 'windows'
```

Comment:
This low-level logic could be replaced by https://docs.pytest.org/en/6.2.x/parametrize.html#pytest-generate-tests